apps/whois/mode/whois.pm

```diff
@@ -0,0 +1,308 @@
+        "critical-days:s"     => { name => 'critical', default => "10:"},
+        "custom-date-regex:s" => { name => 'custom_date_regex', default => 'Expiry Date:\s+(\d{2})-(\d{2})-(\d{4})'}, # Regex for whois outcom
+        "custom-date-map:s"   => { name => 'custom_date_map', default => "DMY"}, # day month year
+    });
+    # This method will execute the command, but will not let me parse the outcome
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::init(%options);
+    # Check domain has only two levels
+    my $domain_levels = split('\.', $self->{option_results}->{domain});
+    if ($domain_levels != 2){
+        $self->{output}->add_option_msg(
+            short_msg => "Wrong domain pattern '" . $self->{option_results}->{domain} . "'. Only DOMAIN.TLD supported");
+        $self->{output}->option_exit();
+    }
+    # Check custom date regex has at least 3 groups 
+    my $MinGroupCount = 3;
+    my $opening_groups = ($self->{option_results}->{custom_date_regex} =~ tr/\(//);
+    if ($opening_groups < $MinGroupCount) {
+        $self->{output}->add_option_msg(short_msg => "Error. At least 3 groups needed on regex. found: " . $opening_groups);
+        $self->{output}->option_exit();
+    }
+    my $closing_groups = ($self->{option_results}->{custom_date_regex} =~ tr/\)//);
+    if ($opening_groups != $closing_groups) {
+        $self->{output}->add_option_msg(short_msg => "Error. count of opening and closing parenthesis differ in regex: " . $opening_groups .'<>'. $closing_groups);
+        $self->{output}->option_exit();
+    }
+    # Check custom date map has same number of groups as the regex, and only contains the Characters YMDhms
+    unless ($self->{option_results}->{custom_date_map} =~ /Y/ && $self->{option_results}->{custom_date_map} =~ /D/ && $self->{option_results}->{custom_date_map} =~ /M/) {
+        $self->{output}->add_option_msg(short_msg => "Error. custom-date-map must contaion Y, M and D characters.");
+        $self->{output}->option_exit();
+    }
+    if (($self->{option_results}->{custom_date_map} =~ tr/YMDhms//) > $opening_groups) {
+        $self->{output}->add_option_msg(short_msg => "custom-date-map expects more groups than the count defined in custom-date-regex");
+        $self->{output}->option_exit();
+    }
+
+
+    if (($self->{perfdata}->threshold_validate(label => 'warning', value => $self->{option_results}->{warning})) == 0) {
+        $self->{output}->add_option_msg(short_msg => "Wrong warning threshold '" . $self->{option_results}->{warning} . "'.");
+        $self->{output}->option_exit();
+    }
+    if (($self->{perfdata}->threshold_validate(label => 'critical', value => $self->{option_results}->{critical})) == 0) {
+        $self->{output}->add_option_msg(short_msg => "Wrong critical threshold '" . $self->{option_results}->{critical} . "'.");
+        $self->{output}->option_exit();
+    }
```

Comment:
YOu will not need this after refactoring on top of counter template class

Reply:
removed